### PR TITLE
Add Login Platforms to settings menu

### LINKED ISSUE #2829: Create a Login settings menu item and view
# User story
As an admin user I need to be able to connect a FIWARE identity manager with an APInf instance by giving required settings

# Description
Admin should be able to give both Github OAuth and FIWARE OAuth login credentials. It could make sense to move the current Github secret settings from Settings view to a Login view and have a separate menu item for Login. (Compare with Proxies).

# Goal
* [ ] Admin can add the required OAuth settings for FIWARE sign in
* [ ] Admin can find this from a logical place together with Github id
* [ ] Admin can easily tell that they are giving details for Login
  * Note: Currently the Github settings for Github login only say Github, they do not mention login

This issue is related to #2767.

Open issue: list of required settings

# Wireframe
In Settings menu from APInf navbar, a new submenu called Login Platform is included.
Under the menu both Fiware and GitHub login credentials can be added..
This is supportive of listing future login and authentication platforms.

image

### PR DESCRIPTION
**Pull request by midyan**
Closes #2829

## To Do

 - [x] Create Login Platform path to Settings package
 - [x] Move Github configuration form to Login Platform path
 - [x] Add Fiware configuration form to Login Platform path
 - [x] Add FIware Login options to Login Screen

### CODE REVIEW

File: apinf_packages/settings/collection/schema.js [hunk 1]
```diff
@@ -130,24 +130,6 @@ Settings.schema = new SimpleSchema({
     regEx: SimpleSchema.RegEx.Email,
     optional: true,
   },
-  githubConfiguration: {
```
Comment (mauriciovieira) [inline apinf_packages/settings/collection/schema.js]: Why? 

I disagree with removal of github configuration from here. 

Why not just add FIWARE config here?

**Comment by mauriciovieira** [comment]
I can't see the reason for this refactoring. It is still a setting, a unique setting for the Platform installation.

**Comment by mauriciovieira** [comment]
Also, for a general rule, use imperative on comments (_Add this_, _Set that_) , instead of 3rd person speech (_Adds this_, _Sets that_). For instance, see https://github.com/apinf/platform/blob/develop/apinf_packages/organizations/client/lib/router.js by @brylie

**Comment by midyan** [comment]
I thought it was better to create a new package for the Login Platforms along with some collections refactoring since a new path will be added (*/settings/login-platforms*), since a few other packages do (*branding, proxies*).
Do you think it would be better to keep the collections unaltered?

**Comment by mauriciovieira** [comment]
Good point. I now think _branding_ and _proxies_ should be under _settings_, but we'll review and refactor this structure later :-)

Alright I'm convinced. Please keep doing this way, following the pattern of _branding_.

File: apinf_packages/core/helper_functions/validate_settings.js [hunk 2]
```diff
@@ -14,6 +14,18 @@ export function githubSettingsValid (settings) {
   return false;
 }
 
+// Validates github configuration settings
```
Comment (mauriciovieira) [inline apinf_packages/core/helper_functions/validate_settings.js]: _Validate_

File: apinf_packages/core/helper_functions/validate_settings.js [hunk 3]
```diff
@@ -14,6 +14,18 @@ export function githubSettingsValid (settings) {
...
+export function fiwareSettingsValid (settings) {
+  if ((typeof settings !== 'undefined') && settings.fiwareConfiguration) {
+    const fiwareConfiguration = settings.fiwareConfiguration;
```
Comment (mauriciovieira) [inline apinf_packages/core/helper_functions/validate_settings.js]: Add a descriptive comment

File: apinf_packages/login_platforms/collection/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/* Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
+
+// Meteor packages imports
+import { Mongo } from 'meteor/mongo';
+
```
Comment (mauriciovieira) [inline apinf_packages/login_platforms/collection/index.js]: Remove extra line

File: apinf_packages/login_platforms/server/methods.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
+
+// Meteor packages imports
+import { Meteor } from 'meteor/meteor';
+
+// Meteor contributed packages imports
+import { ServiceConfiguration } from 'meteor/service-configuration';
+
+// APInf imports
+// eslint-disable-next-line max-len
```
Comment (mauriciovieira) [inline apinf_packages/login_platforms/server/methods.js]: Remove this eslint disable by following [this example](https://github.com/apinf/platform/pull/2820#discussion_r137027507)

**Comment by mauriciovieira** [comment]
Rebase develop
```
git checkout develop
git pull
git checkout feature/fiware_login_settings_menu
git rebase develop
# fix conflicts
git push --force origin feature/fiware_login_settings_menu
```

**Comment by midyan** [comment]
@mauriciovieira All done! Ready for review

**Comment by mauriciovieira** [comment]
Rebase develop _again_

```
git checkout develop
git pull
git checkout feature/fiware_login_settings_menu
git rebase develop
# fix conflicts
git push --force origin feature/fiware_login_settings_menu
```

Also, can you do a interactive rebase and shorten the commit messages?
When github has to use ... it's because it's too big.

You should follow [thoughtbots tips on commit messages](https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message) I already told you.

BTW, if you don't understand, I can do this too, after everything is finished.

**Comment by midyan** [comment]
@mauriciovieira I have added the chimp tests to the new UI elements, as we discussed. I only couldn't figure out how to add the Fiware Icon to the login button. Any ideas?

**Comment by mauriciovieira** [comment]
@midyan I am looking into it now :-) If everything is ok, I'll merge.

**Comment by mauriciovieira** [comment]
@midyan the icon is automatically added by meteor:useraccounts-bootstrap https://github.com/meteor-useraccounts/bootstrap/commit/7f5f96609b7c5bb2e6a5128eb2924c6ea7d9f176#diff-a3f4fe2db2bb0ddf936b0a8d7e3fe304R3 I don't know a solution for it now.

**Pull request merged**